magma/circuit.py

```diff
@@ -926,3 +926,29 @@ def finalize(self, dont_instantiate: bool = False):
         for k, v in self._inst_attrs.items():
             setattr(inst, k, v)
         return inst
+
+
+class DebugCircuitKind(DefineCircuitKind):
```

Comment:
would call it DebugDefineCircuitKind

I actually think we should get rid of the DefineCircuitKind <-> CircuitKind indirection, but that can be later.